updater.go

```diff
@@ -184,6 +182,77 @@ func (u *Updater) update(ctx Context, options UpdateOptions) (*Update, error) {
 	return update, nil
 }
 
+func (u *Updater) ApplyDownloaded(ctx Context) (bool, error) {
+	options := ctx.UpdateOptions()
+
+	// 1. check with the api server again for the latest update to be sure that a new update has not come out since our last call to CheckAndDownload
+	u.log.Infof("Attempting to apply previously downloaded update")
+	update, err := u.checkForUpdate(ctx, options)
+
+	if err != nil {
+		return false, findErr(err)
+	}
+
+	// Only report apply success/failure
+	applied, err := u.applyDownloaded(ctx, update, options)
+	if err != nil && !applied {
+		report(ctx, err, update, options)
+	}
+	if err != nil {
+		return false, err
+	}
+	return applied, nil
+
+}
+
+// ApplyDownloaded will look for an previously downloaded update and attempt to apply it without prompting.
+// CheckAndDownload must be called first so that we have a download asset avaiable to apply.
+func (u *Updater) applyDownloaded(ctx Context, update *Update, options UpdateOptions) (applied bool, err error) {
+	if update == nil || !update.NeedUpdate {
+		u.log.Infof("No previously downloaded update to apply since client is update to date")
+		return false, nil
+	}
+	u.log.Infof("Got update with version: %s", update.Version)
+
+	if u.updateMissingAsset(update) {
+		return false, nil
+	}
+
+	// 2. check the disk via FindDownloadedAsset. Compare our API result to this result. If the downloaded update is stale, clear it and start over.
+	downloadedAssetPath, err := u.FindDownloadedAsset(update.Asset.Name)
+	defer func() {
```

Comment:
this should only happen on a successful download, can you move it below the error check?